File: java/data/src/main/java/com/youknow/data/source/cache/MoviesCacheDataSource.java
package com.youknow.data.source.cache;

import com.youknow.data.source.MoviesDataSource;
import com.youknow.domain.repository.MoviesRepository;

public class MoviesCacheDataSource implements MoviesDataSource {

    @Override
    public void getNowPlaying(MoviesRepository.MoviesLoadedCallback callback) {

    }

    @Override
    public void getUpcoming(MoviesRepository.MoviesLoadedCallback callback) {

    }

    @Override
    public void getMovie(String movieId, MoviesRepository.MovieLoadedCallback callback) {

    }
}

File: kotlin-coroutine-aac/app/src/main/java/com/youknow/movie/data/source/cache/MoviesCacheDataSource.kt
package com.youknow.movie.data.source.cache

import com.youknow.movie.data.source.MoviesDataSource
import com.youknow.movie.domain.model.Movie
import com.youknow.movie.domain.model.SimpleMovie

class MoviesCacheDataSource: MoviesDataSource {

    override suspend fun getNowPlaying(): List<SimpleMovie> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override suspend fun getUpcoming(): List<SimpleMovie> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override suspend fun getMovie(id: String): Movie {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}